### DISCUSSION TITLE
使用身份验证服务登录的方案能否分配用户可用模型？

### DISCUSSION THREAD

**Discussion by BiFangKNT** [lobehub/lobehub#3884] [Self Hosting]
配合Cloudflare Zero Trust的身份验证服务，能否做到分配每个用户的可用模型？比如说我不想让a用户使用glm4plus模型，但是可以使用glm4flash这种免费模型。我看了auth0、CF、logto的文档，似乎这些服务商只能对登录进行限制，而不能限制登录后，在lobe里的多账户的账户权限？

**Comment by arvinxx** [comment]
这个要做账号权限管理了，这个逻辑应该不太会考虑在开源版里做，因为开源版核心面向个人用户。未来做 toB / 团队版的时候会考虑这块的需求

**Reply by BiFangKNT** [reply]
团队版的话是不是还不在日程上？是否一年内都不会考虑？

**Reply by BiFangKNT** [reply]
我觉得即使是简单一点的，不如未来团队版的功能丰富的，也还是有账户权限管理比较合理，毕竟就算是个人用户，也应该有给家人或朋友用的场景，一些比较贵的模型不想给他人用也是合乎常理的需求。

**Reply by arvinxx** [reply]
@BiFangKNT 团队版在 Roadmap 里：https://github.com/orgs/lobehub/projects/5

**Reply by BiFangKNT** [reply]
@arvinxx 话说这种毛玻璃效果能做进本体里吗？
![image](https://github.com/user-attachments/assets/5b920f91-1ee8-448a-8710-9e194375c200)

**Reply by arvinxx** [reply]
@BiFangKNT 我感觉以后在桌面端版本里可以试试，对话界面也加这个估计性能会不太行。参考 #3156

**Reply by BiFangKNT** [reply]
请问桌面端版本和对话界面有什么区别？对话界面是网页端，渲染在服务器上，而桌面端的渲染是调用客户端的？网页端不能调用客户端硬件吗？

**Reply by arvinxx** [reply]
客户端可以是独立的应用进程，有方法可以提升性能的。纯网页里不好弄

**Reply by arvinxx** [reply]
另外纯毛玻璃的效果，macOS 系统就自带，所以可以实现性能开销非常低的毛玻璃效果，感觉起来会非常原生

**Reply by BiFangKNT** [reply]
客户端计划什么时候上线呢？

**Reply by BiFangKNT** [reply]
win10&11呢🤣

**Reply by arvinxx** [reply]
> win10&11呢🤣

windows 我记得也有的

**Reply by arvinxx** [reply]
> 客户端计划什么时候上线呢？

计划 Q4 末开始做，估计明年初吧

**Reply by arvinxx** [reply]
> 毕竟就算是个人用户，也应该有给家人或朋友用的场景，一些比较贵的模型不想给他人用也是合乎常理的需求。

这种其实你现在用 one-api 做 key 维度的模型控制就能实现吧

**Reply by BiFangKNT** [reply]
> > 毕竟就算是个人用户，也应该有给家人或朋友用的场景，一些比较贵的模型不想给他人用也是合乎常理的需求。
> 
> 这种其实你现在用 one-api 做 key 维度的模型控制就能实现吧

意思是给朋友做一个限制模型的key，然后自己用原本的key？行应该是行，但是有点麻烦，还是lobe本体能够实现这个功能最好，应该也不难实现吧。
另外环境变量里指定了openai的代理地址和key，在网页里再设置为别的地址和key的话，会把环境变量的值覆盖吗？能实现我给环境变量设置为one-api阉割过模型的key使之默认是阉割模型的，然后我使用的时候在页面里再填入我的原生key使我能使用完整的模型吗？

**Reply by lai-bluejay** [reply]
或者是配置到自定义的模型里，给自定义模型再部署个类似API manager的服务来做鉴权？？

**Reply by BiFangKNT** [reply]
> 或者是配置到自定义的模型里，给自定义模型再部署个类似API manager的服务来做鉴权？？

我那个时候是没有自定义模型的，现在就很好搞了🤣
